roles/openshift_excluder/tasks/status.yml

```diff
@@ -34,25 +34,37 @@
   - not openshift.common.is_containerized | bool
 
 - name: Record excluder status
-  command: "{{ openshift.common.service_type }}-excluder"
+  command: "{{ openshift.common.service_type }}-excluder status"
   register: excluder_status
   when:
   - "{{ openshift_excluder_installed.installed_versions | default([]) | length > 0 }}"
   - not openshift.common.is_containerized | bool
   failed_when: false
 
 - name: Record docker excluder status
-  command: "{{ openshift.common.service_type }}-docker-excluder"
+  command: "{{ openshift.common.service_type }}-docker-excluder status"
   register: docker_excluder_status
   when:
   - "{{ docker_excluder_installed.installed_versions | default([]) | length > 0 }}"
   - not openshift.common.is_containerized | bool
   failed_when: false
 
+# Unfortunately rpm_q doesn't return installed_versions if there aren't any installed
+#
+# The excluder status function returns 0 when everything is excluded
+# and 1 if any packages are missing from the exclusions list and outputs a warning to stderr
+# # atomic-openshift-excluder status ; echo $?
+# exclude -- All packages excluded
+# 0
+# # atomic-openshift-excluder unexclude
+# # atomic-openshift-excluder status ; echo $?
+# unexclude -- At least one package not excluded
+# 1
+
 - name: Set excluder status facts
   set_fact:
-    docker_excluder_enabled: "{{ 'false' if docker_excluder_status.rc | default(0) == 0 or docker_excluder_installed.installed_versions | default(0) | length == 0 else 'true' }}"
-    openshift_excluder_enabled: "{{ 'false' if docker_excluder_status.rc | default(0) == 0 or openshift_excluder_installed.installed_versions | default(0) | length == 0 else 'true' }}"
+    docker_excluder_enabled: "{{ 'false' if docker_excluder_status.rc | default(0) != 0 or docker_excluder_installed.installed_versions | default(0) | length == 0 else 'true' }}"
+    openshift_excluder_enabled: "{{ 'false' if docker_excluder_status.rc | default(0) != 0 or openshift_excluder_installed.installed_versions | default(0) | length == 0 else 'true' }}"
```

Comment:
Oh, that's the other thing, I intended for these two facts to be used only for determining whether or not the excluders should be re-enabled after the upgrade is complete. I'd expect the upgrade playbooks to disable {{ openshift.common.service_type}}-excluder no matter what and i guess we need to ensure that the upgrade playbooks only upgrade you to the version that they're intended to upgrade you to. ie: playbooks/byo/openshift-cluster/upgrades/v3_5/upgrade.yml should ensure that it doesn't upgrade you to anything but the latest 3.5.

The docker excluder in particular I feel we should leave enabled no matter what. The upgrade playbooks need to update to the correct docker excluder version which will set the proper excludes line and then we perform the actual docker upgrade.